v2/internal/reconcilers/arm/azure_generic_arm_reconciler_instance.go

```diff
@@ -542,3 +544,16 @@ func ConvertToARMResourceImpl(
 	result := genruntime.NewARMResource(typedArmSpec, nil, armID)
 	return result, nil
 }
+
+// MustGetAPIVersion returns the ARM API version for the resource we're reconciling
+func (r *azureDeploymentReconcilerInstance) MustGetAPIVersion() string {
+	metaObject := r.Obj
+	scheme := r.ResourceResolver.Scheme()
+
+	apiVersion, err := genruntime.GetAPIVersion(metaObject, scheme)
+	if err != nil {
+		panic(err)
```

Comment:
I think we should avoid `panic` in the reconciler codepaths as much as we can? I know it makes it more awkward to use but can we return `err` here instead?

Reply:
Agree with @matthchr here.

Reply:
I was on the fence; originally wrote it with errors and then changed to use a panic to reduce boilerplate.

Changed it back.